modules/sops/default.nix

```diff
@@ -318,6 +332,12 @@ in {
             builtins.isPath secret.sopsFile ||
             (builtins.isString secret.sopsFile && lib.hasPrefix builtins.storeDir secret.sopsFile);
           message = "'${secret.sopsFile}' is not in the Nix store. Either add it to the Nix store or set sops.validateSopsFiles to false";
+        } {
+          assertion = secret.owner != "root" && secret.uid != null || secret.owner != "root" && secret.uid == null;
+          message = "In ${secret.name} exactly one of sops.owner and sops.uid must be set";
+        } {
+          assertion = secret.group == "" && secret.gid != null || secret.group != "" && secret.gid == null;
```

Comment:
```suggestion
          assertion = secret.group != "root" -> secret.gid != null;
```

Reply:
Shouldn't this be `assertion = secret.group != users.${secret.owner}.group -> secret.gid == null;` then? I'm pretty new to Nix, so I'm not sure if I understand this 100%.

Reply:
I think you are right.